### PR TITLE
[ML][Pipelines]Fix if_else CLI error and add related tests

### PR DESCRIPTION
**Pull request by D-W-**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_builders/condition_node.py [hunk 1]
```diff
@@ -63,16 +69,37 @@ def _validate_params(self, raise_error=True) -> MutableValidationResult:
...
             )
-        if self.false_block is not None and not isinstance(self.false_block, (BaseNode, AutoMLJob)):
+        if self.false_block is not None and not isinstance(self.false_block, (BaseNode, AutoMLJob, str)):
             validation_result.append_error(
                 yaml_path="false_block", message=error_msg.format("false_block", type(self.false_block))
             )
 
+        # check if true/false block is valid binding
+        for name, block in {"true_block": self.true_block, "false_block": self.false_block}.items():
+            if block is None or not isinstance(block, str):
```
Comment (lllilithyang) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_builders/condition_node.py:94]: Can we put the above lines in this for? As the above lines are both built on 'name' and 'block'.

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_builders/condition_node.py [hunk 2]
```diff
@@ -63,16 +69,37 @@ def _validate_params(self, raise_error=True) -> MutableValidationResult:
...
                 yaml_path="true_block", message=error_msg.format("true_block", type(self.true_block))
             )
-        if self.false_block is not None and not isinstance(self.false_block, (BaseNode, AutoMLJob)):
+        if self.false_block is not None and not isinstance(self.false_block, (BaseNode, AutoMLJob, str)):
             validation_result.append_error(
                 yaml_path="false_block", message=error_msg.format("false_block", type(self.false_block))
             )
 
+        # check if true/false block is valid binding
+        for name, block in {"true_block": self.true_block, "false_block": self.false_block}.items():
```
Comment (lllilithyang) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_builders/condition_node.py:93]: Can we put the above lines in this for? As the above lines are both built on 'name' and 'block'.

File: sdk/ml/azure-ai-ml/tests/pipeline_job/e2etests/test_control_flow_pipeline.py [hunk 3]
```diff
@@ -44,6 +54,86 @@ class TestConditionalNodeInPipeline(AzureRecordedTestCase):
     pass
 
 
+class TestIfElse(TestConditionalNodeInPipeline):
+    def test_happy_path_if_else(self, client: MLClient, randstr: Callable[[], str]) -> None:
```
Comment (lllilithyang) [inline sdk/ml/azure-ai-ml/tests/pipeline_job/e2etests/test_control_flow_pipeline.py:58]: May combine test_happ_path_if_else, test_if_else_one_branch and test_if_else_one_branch into one test, using `@pytest.mark.parametrize`.

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_builders/condition_node.py (hunk 2 above)
Comment (D-W-) [reply sdk/ml/azure-ai-ml/azure/ai/ml/entities/_builders/condition_node.py:93]: Above check is not exactly same with this one with different binding check (${{parent.jobs.xxx.outputs.output}}, ${{parent.jobs.xxx}}) and exception message. Merge them may bring more complexity.

File: sdk/ml/azure-ai-ml/tests/pipeline_job/e2etests/test_control_flow_pipeline.py (hunk 3 above)
Comment (D-W-) [reply sdk/ml/azure-ai-ml/tests/pipeline_job/e2etests/test_control_flow_pipeline.py:58]: Currently in e2e test, each parametrized test case will become a file with long file name, which need to take more steps to handle. Reference `test_pipeline_job_with_command_job` in `pipeline_job/e2etests` as an example. I think split tests makes it clearer compared to use the fixture.

File: sdk/ml/azure-ai-ml/azure/ai/ml/_schema/pipeline/pipeline_component.py [hunk 4]
```diff
@@ -107,6 +108,10 @@ def _post_load_pipeline_jobs(context, data: dict) -> dict:
                     loaded_data=job_instance,
                 )
                 jobs[key] = job_instance
+            elif job_instance.get("type") == ControlFlowType.IF_ELSE:
+                # Convert to if-else node.
+                job_instance = ConditionNode._create_instance_from_schema_dict(loaded_data=job_instance)
+                jobs[key] = job_instance
```
Comment (wangchao1230) [inline sdk/ml/azure-ai-ml/azure/ai/ml/_schema/pipeline/pipeline_component.py:114]: seems like many duplicate line:

jobs[key] = job_instance

**Review by wangchao1230 — APPROVED** [review]

File: sdk/ml/azure-ai-ml/azure/ai/ml/_schema/pipeline/pipeline_component.py (hunk 4 above)
Comment (D-W-) [reply sdk/ml/azure-ai-ml/azure/ai/ml/_schema/pipeline/pipeline_component.py:114]: Good catch! Will fix this in next PR

**Pull request merged**